plugins/modeltraining/pkg/modeltraining/api.go

```diff
@@ -0,0 +1,151 @@
+package modeltraining
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	modeltraining "github.com/rancher/opni/plugins/modeltraining/pkg/apis/modeltraining"
+	"google.golang.org/protobuf/types/known/emptypb"
+	k8scorev1 "k8s.io/api/core/v1"
+)
+
+type ModelTrainingParameters struct {
+	ByCluster map[string]*ClusterParameters `json:",inline,omitEmpty"`
+}
+
+type ClusterParameters struct {
+	ByNamespace map[string]*NamespaceParameters `json:",inline,omitEmpty"`
+}
+
+type NamespaceParameters struct {
+	ByDeployment map[string]*Deployment `json:",inline,omitEmpty"`
+}
+
+type Deployment struct {
+	DeploymentName string `json:",inline,omitEmpty"`
+}
+
+func (c *ModelTrainingPlugin) TrainModel(ctx context.Context, in *modeltraining.WorkloadsList) (*corev1.Reference, error) {
+	var modelTrainingParameters = map[string]map[string][]string{}
+	for idx := 0; idx < len(in.List); idx++ {
```

Comment:
The correct way to enumerate a list in go is
```
for index, item := range in.List {}
```
For what you're doing you don't actually need the index either because you just want to operator on each item so you can do
```
for _, item := range in.List {}
```